roachtest: pass RunOptions instead of NodeListOption

herkolategan opened the pull request:
This change updates the `Run` and `RunE` cluster interface methods to take `install.RunOptions` as a parameter instead of `option.NodeListOption` to expose the additional available options to `roachtest`.

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/117320)
<!-- Reviewable:end -->

herkolategan reviewed (commented):
TC Run: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/13396744

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117320#-:-NnPPs9pBI5tu5VXg-fJ:b8jfh7n)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs and @srosenberg)



<!-- Sent from Reviewable.io -->

srosenberg commented on pkg/cmd/roachtest/option/options.go:
```diff
@@ -80,3 +80,8 @@ type StopOpts struct {
 }
+
+// OnNodes returns a RunOptions that will run on the given nodes.
+func OnNodes(nodes NodeListOption) install.RunOptions {
```
Nit: `WithNodes` seems more standard

srosenberg commented on pkg/cmd/roachtest/cluster/cluster_interface.go:
```diff
@@ -101,11 +101,11 @@ type Cluster interface {
-	Run(ctx context.Context, node option.NodeListOption, args ...string)
+	Run(ctx context.Context, options install.RunOptions, args ...string)
 
 	// RunE runs a command on the specified nodes and returns an error.
```
Nit: might be worth expanding the comment; e.g.,

> ... on the specified nodes (`RunOptions.Nodes`) with specified `RetryOptions`...

herkolategan committed:
roachprod: add WithNodes convenience function

Previously, OnNodes was added to roachprod as a method to easily create
RunOptions that will run on the specified nodes. This function has been renamed
to WithNodes to be more aligned with the other RunOption methods.

Additionally, a convenience function, WithNodes, has been added to roachtest
options to instantiate RunOptions with a NodeListOption.

Epic: None
Release Note: None

herkolategan committed:
roachtest: pass RunOptions instead of NodeListOption

This change updates the `Run` and `RunE` cluster interface methods to take
`install.RunOptions` as a parameter instead of `option.NodeListOption` to expose
the additional available options to `roachtest`.

Epic: None
Release Note: None

herkolategan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117320)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @renatolabs)

___
*[`pkg/cmd/roachtest/cluster/cluster_interface.go` line 106 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117320#gh-1445354087:-NnmrFeFEbfvJUM7cqqS:b-gh93gd) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f71972a72df4ea0d589ab48c4d954b42d59945a/pkg/cmd/roachtest/cluster/cluster_interface.go#L106)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Nit: might be worth expanding the comment; e.g.,

> ... on the specified nodes (`RunOptions.Nodes`) with specified `RetryOptions`...
</blockquote></details>

Good idea, Done.
___
*[`pkg/cmd/roachtest/option/options.go` line 85 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117320#gh-1445351068:-Nnmqu2_5PhQ92IB_7wH:bv3cuy) ([raw file](https://github.com/cockroachdb/cockroach/blob/5f71972a72df4ea0d589ab48c4d954b42d59945a/pkg/cmd/roachtest/option/options.go#L85)):*
<details><summary><i>Previously, srosenberg (Stan Rosenberg) wrote…</i></summary><blockquote>

Nit: `WithNodes` seems more standard
</blockquote></details>

Makes sense, I'll update the roachprod side to`WithNodes` as well for consistency.


<!-- Sent from Reviewable.io -->

herkolategan commented:
TFTR!

bors r=srosenberg
